index.js

```diff
@@ -48,6 +48,44 @@ class versionRequest {
     }
   }
 
+  static setVersionByAcceptHeader (customFunction) {
+    function removeWhitespaces (str) {
+      return str.replace(/\s/g, '')
+    }
+
+    return (req, res, next) => {
+      if (req && req.headers && req.headers.accept) {
+        if (customFunction && typeof customFunction === 'function') {
+          req.version = customFunction(req.headers.accept)
+          if (typeof req.version !== 'string') {
+            req.version = this.isObject(req.version) ? JSON.stringify(req.version) : req.version.toString()
+          }
+        } else {
+          const params = req.headers.accept.split(';')[1]
+          const paramMap = {}
+          if (params) {
+            for (let i of params.split(',')) {
+              const keyValue = i.split('=')
+              paramMap[removeWhitespaces(keyValue[0]).toLowerCase()] = removeWhitespaces(keyValue[1])
+            }
+            req.version = paramMap.version
+          }
+
+          if (req.version === undefined) {
+            const header = removeWhitespaces(req.headers.accept)
```

Comment:
maybe we can put this logic for the alternative format in a separate, new function? it makes it easier to then just unit that that alone.

I'm just as bad as the next guy at naming things, but maybe we call it `setVersionByAcceptFormat` :-) but I'll let you choose